packages/esm-patient-chart-app/src/clinical-views/utils/helpers.ts

```diff
@@ -0,0 +1,189 @@
+  return allObs?.length == 1 ? allObs[0] : allObs?.sort(obsArrayDateComparator)[0];
+}
+
+export function getObsFromEncounters(encounters: Encounter, obsConcept: string) {
+  const filteredEnc = encounters?.find((enc) => enc.obs.find((obs) => obs.concept.uuid === obsConcept));
+  return getObsFromEncounter(filteredEnc, obsConcept);
+}
+
+export function resolveValueUsingMappings(encounter: Encounter, concept: string, mappings) {
+  const obs = findObs(encounter, concept);
+  for (const key in mappings) {
+    if (typeof obs?.value === 'object' && 'uuid' in obs.value && mappings[key] === obs?.value?.uuid) {
+      return key;
+    }
+  }
+  return '--';
+}
+
+export function getConditionalConceptValue(encounter: Encounter, conditionalConceptMappings, isDate: Boolean) {
+  const { trueConcept, nonTrueConcept, dependantConcept, conditionalConcept } = conditionalConceptMappings;
+  const dependantValue = findObs(encounter, dependantConcept)?.value;
+  const dependantUuid =
+    typeof dependantValue === 'object' && 'uuid' in dependantValue ? dependantValue.uuid : undefined;
+  const finalConcept = dependantUuid === conditionalConcept ? trueConcept : nonTrueConcept;
+  return getObsFromEncounter(encounter, finalConcept, isDate);
+}
+
+export function getConceptFromMappings(encounter: Encounter, concepts: Array<string>) {
+  for (const concept of concepts) {
+    const obs = findObs(encounter, concept);
+    if (obs && obs.value) {
+      return concept;
+    }
+  }
+  return null;
+}
+
+export function getMultipleObsFromEncounter(encounter: Encounter, obsConcepts: Array<string>) {
+  let observations = [];
+  obsConcepts.map((concept) => {
+    const obs = getObsFromEncounter(encounter, concept);
+    if (obs !== '--') {
+      observations.push(obs);
+    }
+  });
+
+  return observations.length ? observations.join(', ') : '--';
+}
+
+export function getObsFromEncounter(
+  encounter: Encounter,
+  obsConcept: string,
+  isDate?: Boolean,
+  isTrueFalseConcept?: Boolean,
+  type?: string,
+  fallbackConcepts?: Array<string>,
+  secondaryConcept?: string,
+  t?: TFunction,
+) {
+  let obs = findObs(encounter, obsConcept);
+  if (!encounter || !obsConcept) {
+    return '--';
+  }
+
+  if (isTrueFalseConcept) {
+    if (typeof obs?.value === 'object') {
+      if (
+        (obs?.value?.uuid != esmPatientChartSchema.trueConceptUuid._default && obs?.value?.name?.name !== 'Unknown') ||
+        obs?.value?.name?.name === t('FALSE')
+      ) {
+        return t('No');
+      } else if (obs?.value?.uuid == esmPatientChartSchema.trueConceptUuid._default) {
+        return t('Yes');
+      } else {
+        return obs?.value?.name?.name;
+      }
+    }
+  }
+
+  if (type === 'location') {
+    return encounter.location.display;
+  }
+
+  if (type === 'provider') {
+    return encounter.encounterProviders.map((p) => p.provider.name).join(' | ');
+  }
+
+  if (type === 'visitType') {
+    return encounter.encounterType.name;
+  }
```

Comment:
All of these seem like a mixture of concerns that should be moved somewhere else, since these are not about obs from an encounter, but properties of the encounter itself.

Reply:
I have added another function to address this